typescript/sdk/src/crud/AbstractCrudModule.ts

```diff
@@ -0,0 +1,61 @@
+import { Logger } from 'pino';
+
+import { Address } from '@hyperlane-xyz/utils';
+
+import { HyperlaneFactories } from '../contracts/types.js';
+import { CoreConfig } from '../core/types.js';
+import { HookConfig } from '../hook/types.js';
+import { IsmConfig } from '../ism/types.js';
+import { ChainMetadataManager } from '../metadata/ChainMetadataManager.js';
+import { WarpRouteConfig } from '../metadata/warpRouteConfig.js';
```

Comment:
may need a diff type going by https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/3671#discussion_r1580905682

Reply:
going to avoid constraining types here and let the implementations pass them up